File: airport_dir/lib/airport.rb
require_relative 'plane'
require_relative 'weather'

class Airport
  include Weather
  DEFAULT_CAPACITY = 20
  def initialize(capacity = DEFAULT_CAPACITY)
    @airport_hanger = []
    @capacity = capacity
  end

  def airport_hanger
    @airport_hanger
  end

  def capacity
    @capacity
  end

end

File: JavaScriptAirport/src/airport.js
'use strict'
class Airport {
  constructor(weather) {
    this.DEFAULT_CAPACITY = 20;
    this._weather = typeof weather !== 'undefined' ? weather : new Weather();
    this._runway = [];
    this.capacity = this.DEFAULT_CAPACITY;
  };

  planes() {
    return this._runway;
  };

  clearForLanding(plane) {
    if (this._weather.isStormy()) {
      throw new Error('Landing prohibited due to storm conditions');
    };
    if (this._runway.length  == this.capacity) {
      throw new Error("Airport Full!");
    };
    this._runway.push(plane);
  };

  clearForTakeoff(plane) {
    if (this._weather.isStormy()) {
      throw new Error('Takeoff prohibited due to storm conditions');
    };
    if (this._runway.length == 0) {
      throw new Error("Airport Empty!");
    };

    this._runway.pop(plane);
  };
  isStormy(){
    return false;
  };
};
